### DISCUSSION TITLE
Adding quantities and units for fluid flow

### DISCUSSION THREAD

**Discussion by sveipa** [angularsen/UnitsNet#1032] [Ideas]
Hi,
ref the https://github.com/angularsen/UnitsNet/wiki/Adding-a-New-Unit - as "is widely used" is a relative concept, I want to discuss whether quantities and units around
- Permeability - https://en.wikipedia.org/wiki/Permeability_(Earth_sciences)
- Compressibility - https://en.wikipedia.org/wiki/Compressibility
would be welcomed to the project?

It is highly used in several domains, like petroleum.

Best regards,
Frank

**Comment by tmilnthorp** [comment]
I think these are great additions.

I have added #1037 for compressibility. Please review.

**Comment by tmilnthorp** [comment]
I do wonder for permeability, since the unit strings are the same as area, what do we do about parsing collisions? @angularsen

**Reply by angularsen** [reply]
Area.Parse("1 m²") and Permeability.Parse("1 m²") should not be problematic, since the abbreviations are defined per quantity. @tmilnthorp

**Comment by sveipa** [comment]
Great, and thanks for the quick replies! One thing to note about permeability is that it is used for different quantities depending on the type of science involved. I see there's one already implemented for electromagnetism using "henry" as unit - https://en.wikipedia.org/wiki/Permeability_(electromagnetism) 

What's the best practice to differentiate between them in UnitsNet? One might be to use different suffixes, like PermeabilityElectromagnetism, PermeabilityFluidMechanics etc.

**Reply by angularsen** [reply]
Yes, we would have to distinguish them by name.

No particular convention, but we have had to distinguish similarly for `UsGallon` and `ImperialGallon`.
I would try to choose whatever is intuitive and preferably see how others refer to them to disambiguate, like this wiki. https://en.wikipedia.org/wiki/Permeability

I would suggest something like ElectromagneticPermeability, FluidMechanicalPermeability, NauticalPermeability. It doesn't really matter, as long as it is relatively short, reads well and intellisense can pick it up as you start typing.

**Comment by sveipa** [comment]
For permeability (fluid mechanics) these units are used a lot in the oil & gas domain:
* md - millidarcy
* d - darcy
* m2 - square meters
* µm2 - square micrometer, square micron

**Comment by tmilnthorp** [comment]
I can add those sure. I think FluidPermeability is concise if everyone agrees?

**Comment by angularsen** [comment]
That seems like a good compromise.
